### PR TITLE
Add liveness probe sidecar container

### PR DESCRIPTION
**Pull request by pkalever**
# Describe what this PR does #
The CSI livenessprobe is a sidecar container that monitors the health of the CSI driver and reports it to Kubernetes via the Liveness Probe mechanism. This enables Kubernetes to automatically detect issues with the driver and restart the pod to try and fix the issue.

Fixes: #1096

### CODE REVIEW

**Comment by pkalever** [comment]
@Mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio rebase

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml [hunk 1]
```diff
@@ -148,6 +148,33 @@ spec:
...
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: liveness-port
+            initialDelaySeconds: 10
+            timeoutSeconds: 3
+            periodSeconds: 2
+            failureThreshold: 5
+        - name: liveness-probe
+          image: quay.io/k8scsi/livenessprobe:canary
```
Comment (pkalever) [inline deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:164]: @Madhu-1 If I remember it right, we wanted to leverage liveness within cephcsi (cephcsi --type liveness) with a different metricspath (may be /healthz) and a different port (may be 8080)

And the  liveness code is complete, this PR should be focusing on yaml changes only.

correct ?

Comment (Madhu-1) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:164]: Yes correct and we need to expose /healthz endpoint

Comment (pkalever) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:164]: Which has to be taken care in the examples/service-monitor.yaml ?

Comment (pkalever) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:164]: ignore, I was looking at a wrong place

Comment (pkalever) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:164]: Have opened a New PR, Please check: https://github.com/ceph/ceph-csi/pull/1665

**Pull request closed without merging**